src/priority_fee.rs

```diff
@@ -275,27 +340,59 @@ impl PriorityFeeTracker {
         }
     }
 
+
+
+    // TODO: DKH - both algos should probably be in some enum (like algo1, algo2) and be passed to
+    // this method instead of sending a bool flag. I'll refactor this in next iteration. already too many changes
     pub fn get_priority_fee_estimates(
         &self,
         accounts: Vec<Pubkey>,
         include_vote: bool,
-        lookback_period: Option<usize>,
+        lookback_period: Option<u32>,
+        calculation1: bool
     ) -> MicroLamportPriorityFeeEstimates {
-        let start = std::time::Instant::now();
+        let start = Instant::now();
+        let micro_lamport_priority_fee_estimates = if calculation1
+        {
+            self.calculation1(&accounts, include_vote, &lookback_period)
```

Comment:
moved calculations to internal methods so I could call them from the sampling thread without skewing the call time as well as reading the values back to the queue.